phpBB/phpbb/controller/helper.php

```diff
@@ -212,12 +212,29 @@ public function error($message, $code = 500)
 	public function message($message, array $parameters = array(), $title = 'INFORMATION', $code = 200)
 	{
 		array_unshift($parameters, $message);
+		$message_text = call_user_func_array(array($this->user, 'lang'), $parameters);
+		$message_title = $this->user->lang($title);
+
+		if ($this->request->is_ajax())
+		{
+			global $refresh_data;
+
+			$json_response = new \phpbb\json_response;
```

Comment:
`return new JsonResponse()` please

our old `\phpbb\json_response` is to be used with `trigger_error()` only (it breaks the flow by calling exit)